File: frontend/src/libs/sentry.ts
import * as Sentry from "@sentry/react";

const APP_ENV = import.meta.env.VITE_SENTRY_ENV ?? "localhost";

Sentry.init({
  enabled: APP_ENV !== "localhost",
  dsn: "https://af70207f31f4d421138d4c4439dad10e@o4507856054779904.ingest.us.sentry.io/4507856063234048",
  environment: APP_ENV,
  integrations: [
    Sentry.browserTracingIntegration(),
    Sentry.browserProfilingIntegration(),
    Sentry.replayIntegration(),
  ],
  tracesSampleRate: 1.0,
  tracePropagationTargets: [
    import.meta.env.VITE_API_BASE_URL,
    import.meta.env.VITE_ADMINISTRATOR_API_BASE_URL,
  ],
  profilesSampleRate: 1.0,
  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 1.0,
});

File: backend/api/libs/sentry.py
import os

import sentry_sdk
from sentry_sdk.integrations.fastapi import FastApiIntegration
from sentry_sdk.integrations.starlette import StarletteIntegration

SENTRY_ENV = os.environ.get("SENTRY_ENV", "localhost")


def init_sentry():
    if SENTRY_ENV == "localhost":
        return
    sentry_sdk.init(
        dsn="https://9602bed59ab7a8d88d8f9c1ae976416d@o4507856054779904.ingest.us.sentry.io/4507856511696896",
        environment=SENTRY_ENV,
        traces_sample_rate=1.0,
        profiles_sample_rate=1.0,
        integrations=[
            StarletteIntegration(transaction_style="url"),
            FastApiIntegration(transaction_style="url"),
        ],
    )
